src/gui/dialogs/migrate_version_selection.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+   Copyright (C) 2008 - 2018 by Jörg Hinrichs <joerg.hinrichs@alice-dsl.de>
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/migrate_version_selection.hpp"
+
+#include "addon/manager_ui.hpp"
+#include "filesystem.hpp"
+#include "game_version.hpp"
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/dialogs/migrate_version_selection.hpp"
+#include "gui/widgets/listbox.hpp"
+#include "gui/widgets/window.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+static lg::log_domain log_version_migration{"gui/dialogs/migrate_version_selection"};
+#define ERR_LOG_VERSION_MIGRATION   LOG_STREAM(err,   log_version_migration)
+#define WRN_LOG_VERSION_MIGRATION   LOG_STREAM(warn,  log_version_migration)
+#define LOG_LOG_VERSION_MIGRATION   LOG_STREAM(info,  log_version_migration)
+#define DBG_LOG_VERSION_MIGRATION   LOG_STREAM(debug, log_version_migration)
+
+namespace gui2::dialogs
+{
+REGISTER_DIALOG(migrate_version_selection)
+
+void migrate_version_selection::execute()
+{
+    migrate_version_selection mig = migrate_version_selection();
+    if(mig.versions_.size() > 0)
+    {
+        mig.show();
+    }
+}
+
+migrate_version_selection::migrate_version_selection()
+{
+    version_info current_version = game_config::wesnoth_version;
+    std::string current_version_str = std::to_string(current_version.major_version())+"."+std::to_string(current_version.minor_version());
+
+    for(unsigned int i = 1; i < current_version.minor_version(); i++)
+    {
+        std::string previous_version_str = std::to_string(current_version.major_version())+"."+std::to_string(current_version.minor_version()-i);
+        std::string previous_addons_dir = boost::replace_all_copy(filesystem::get_addons_dir(), current_version_str, previous_version_str);
+
+        if(filesystem::file_exists(previous_addons_dir))
```

Comment:
Needs a sanity check that `previous_addons_dir != filesystem::get_addons_dir()`.

That can happen with the command-line options `--userconfig-dir` `--userdata-dir`. I can't remember if there's any platform where the default userconfig-dir doesn't have the version number in.